### DISCUSSION TITLE
Cannot "remix-serve" because REMIX_DEV_HTTP_ORIGIN not set

### DISCUSSION THREAD

**Discussion by vrn-dev** [remix-run/remix#7338] [Q&A] [answered]
Hi, I have built an app on remix 1.19.3. Once I ran `pnpm remix build` I got the build folders with no problems. But when i run `pnpm start` or `remix-serve build` I get the following error.

```bash
> remix-serve build

Remix App Server started at http://localhost:3000 (http://192.168.0.154:3000)

/xxx/node_modules/.pnpm/@remix-run+server-runtime@1.19.3/node_modules/@remix-run/server-runtime/dist/dev.js:17
  if (!origin) throw Error("Dev server origin not set");
                     ^
Error: Dev server origin not set
    at Object.broadcastDevReady (/xxx/node_modules/.pnpm/@remix-run+server-runtime@1.19.3/node_modules/@remix-run/server-runtime/dist/dev.js:17:22)
```
Which basically means that `process.env.REMIX_DEV_HTTP_ORIGIN` is not set. Can someone let me know what this variable is?

It also seems my build command is keeping my `NODE_ENV=development`

```bash
> remix build

 info  building... (NODE_ENV=development)
 info  built (1.3s)
```

Can someone please point out where to look to check what has gone wrong.

**Comment by vrn-dev** [comment, marked as answer]
Had to set env variables correctly as they were not being picked up by remix in production mode.

**Reply by amitaggarwal1** [reply]
Seeing the same issue. Can you elaborate on what variables you needed to set

**Reply by gdayton** [reply]
Seeing this same issue as well, setting: `REMIX_DEV_ORIGIN` to `localhost`, and this does not work. Should this env var be set automatically somewhere that may not be running on my end properly?

**Reply by gdayton** [reply]
Followed this v1 -> v2 remix guide (I'm on 2.0.1) and was able to resolve my issue: [Upgrading to V2](https://remix.run/docs/en/main/start/v2). Specifically changing my `dev` script to running `remix dev --manual -c 'node ./server.js'` did the trick (make sure you get the build path right).

**Reply by marcioluis** [reply]
I was getting this error running it in Docker.

In Remix. Anything but `NODE_ENV=production` is resolved as development so it will setup a development environment with hmr, reload, etc.
That is why running with `--manual` solves, it disables all that.

I fixed changing to `NODE_ENV=production`. `remix-serve ./build/index.js`
